### ISSUE TITLE
About the usage of nn.Embedding

### ISSUE THREAD

**Issue by SYTMTHU** [bamtercelboo/pytorch_word2vec#1]
Hi,

When you input a list of indices to an nn.Embedding object, it will return a matrix of (N, W). As sparse = 1, it actually calls torch.index_select to form the output, thus they do not share the same memory as the corresponding vectors in the nn.Embedding object. 

Then, is it still correct to call backward on the loss that was based on the return vectors? Could you please show the test result on your output word vectors?
